airflow/providers/odbc/hooks/odbc.py

```diff
@@ -211,3 +211,10 @@ def get_sqlalchemy_connection(
         engine = self.get_sqlalchemy_engine(engine_kwargs=engine_kwargs)
         cnx = engine.connect(**(connect_kwargs or {}))
         return cnx
+
+    @staticmethod
+    def _make_serializable(result: list[pyodbc.Row]) -> list[tuple]:
+        """Transform the pyodbc.Row objects returned from a SQL command into
+        JSON-serializable objects.
+        """
+        return [tuple(row) for row in result]
```

Comment:
I think this would be a breaking change for existing users, they would expect the return value to be `list[pyodbc.Row]` and not `list[tuple]`. Maybe we can change the type based on a bool flag?

Reply:
Indeed, I added a flag which disable the serializable data structure by default.